### PR TITLE
Refactor main docks toggle and integration and browser docks

### PR DESCRIPTION
**Pull request by tytan652**
### Description

**Dependency of:**
- https://github.com/obsproject/obs-studio/pull/7637
- https://github.com/obsproject/obs-studio/pull/7638

Groundwork for other PRs and also avoid creating QActions by using the one included in the dock by modifying its connections making it usable even while the docks are locked.

### Motivation and Context

Preparation for PRs that depends on this one.

### How Has This Been Tested?

- Test docks toggle with docks locked or not
- Connect to services integration
- Modify custom browser docks

### Types of changes

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: UI/window-extra-browsers.cpp [hunk 1]
```diff
@@ -234,11 +233,13 @@ void ExtraBrowsersModel::Apply()
...
 	for (int i = deleted.size() - 1; i >= 0; i--) {
 		int idx = deleted[i];
-		main->extraBrowserDockActions.removeAt(idx);
 		main->extraBrowserDockTargets.removeAt(idx);
 		main->extraBrowserDocks.removeAt(idx);
 	}
 
+	if (main->extraBrowserDocks.empty())
+		main->extraBrowserMenuDocksSeparator.clear();
+
```
Comment (WizardCM) [inline UI/window-extra-browsers.cpp:242]: If the user doesn't initially have browser docks, I'm not seeing a place where this gets inserted when the user creates their first custom browser dock.

Comment (tytan652) [reply UI/window-extra-browsers.cpp:242]: Here it the the new `AddDockWidget` function:

https://github.com/obsproject/obs-studio/pull/7636/files#diff-4c0a52244f264feca9c3031c90edcbbbd1087a3040c15f9df4ca9700c5154fa3R9910-R9922

Comment (WizardCM) [reply UI/window-extra-browsers.cpp:242]: Ah, you're right, I'm just blind.

Comment (tytan652) [reply UI/window-extra-browsers.cpp:242]: I need to add the check for separator to the old one too, to add the dock before the separator 🤦.

**Comment by tytan652** [comment]
Old docks are now added before the separator if it exist.

**Review by WizardCM — APPROVED** [review]
Tested as heavily as I could, with a combination of 

* custom browser docks, service integration via Twitch, YouTube and Restream
* docking, undocking
* switching scene collections & profiles
* linking & unlinking service integration

Looks good to merge to me.

**Pull request merged**